### PR TITLE
Fix ServiceIndexResourceV3Provider tests in MacOS Mojave

### PR DESCRIPTION
**Pull request by PatoBeltran**
## Bug

Fixes some tests that fail on MacOS Mojave. For some reason xunit seems to be caching the the `ServiceIndexResourceV3Provider` so when multiple tests use the same source url the test fails. This PR makes sure all the source url used on these tests are different.

**Note:** Our mac CI runs on MacOS Sierra, therefore this tests are not currently failing on CI and this PR shouldn't affect the current state of the tests in any other platform

### CODE REVIEW

File: test/NuGet.Core.Tests/NuGet.Protocol.Tests/ServiceIndexResourceV3ProviderTests.cs [hunk 1]
```diff
@@ -250,7 +251,7 @@ public async Task Query_For_Resource_With_Precedence()
 
             var endpoints = resource.GetServiceEntryUris(new NuGetVersion(4, 0, 0), "Chocolate", "Vegetable");
 
-            Assert.True(endpoints.Count == 1);
+            // Assert.True(endpoints.Count == 1);
```
Comment (nkolev92) [inline test/NuGet.Core.Tests/NuGet.Protocol.Tests/ServiceIndexResourceV3ProviderTests.cs]: why did you remove this?

Comment (PatoBeltran) [reply test/NuGet.Core.Tests/NuGet.Protocol.Tests/ServiceIndexResourceV3ProviderTests.cs]: This should be reverted.

**Review by jainaashish — APPROVED** [review]
👏

**Review by nkolev92 — APPROVED** [review]
